app/scripts/move-popup/dimMoveItemProperties.directive.js

```diff
@@ -186,5 +154,69 @@
         });
       }
     }
+
+    /*
+     * Determins if the item has damange and it's non kinetic.
+     *
+     * @param {Object} item - A item from the destiny server
+     * @return {Boolean}
+     */
+    function itemHasNonKineticDamage(item) {
+      if (!item.dmg || item.dmg === 'kinetic') {
+        return false;
+      }
+      return true;
+    }
+
+    /*
+     * Check for a lockable or elemental property in an item.
+     *
+     * @param {Object} item - A item from the destiny server
+     * @return {Boolean}
+     */
+    function asideHasLockOrElement(item) {
+      if (item.lockable) {
+        return true;
+      }
+      if (item.dmg) {
+        return true;
+      }
+      return false;
+    }
+
+    /*
+     * Returns if the details are currently visible
+     *
+     * @return {Boolean}
+     */
+    function detailsVisible() {
+      if (!vm.showDetailsByDefault && (vm.showDescription || vm.hasDetails) && !vm.item.classified) {
+        return true;
+      }
+      return false;
+    }
+
+    /*
+     * Returns if the type should show completion.
+     *
+     * @param {string} type - The item type
+     * @return {Boolean}
+     */
+    function typeShowsCompletion(type) {
+      var typesShowingCompletion = [
+        'Bounties',
+        'Quests'
+      ];
+      var typesShowingCompletionLength = typesShowingCompletion.length
```

Comment:
Rather than hardcoding this, we can detect it by looking at the `objectives` on the item.